api/tests/integration/infrastructure/repositories/account-recovery-demand-repository_test.js

```diff
@@ -47,38 +46,54 @@ describe('Integration | Infrastructure | Repository | account-recovery-demand-re
       });
 
       context('when demand is unused yet', () => {
-        const email = 'someMail@example.net';
-        let demandId;
 
-        beforeEach(() => {
-          demandId = databaseBuilder.factory.buildAccountRecoveryDemand({ email, temporaryKey, used: false }).id;
+        it('should return the account recovery demand when temporary key not expired', async () => {
+          // given
+          const email = 'someMail@example.net';
+          const temporaryKey = 'someTemporaryKey';
+          const demandId = databaseBuilder.factory.buildAccountRecoveryDemand({ email, temporaryKey, used: false }).id;
           databaseBuilder.factory.buildAccountRecoveryDemand({ email, used: false });
-          return databaseBuilder.commit();
-        });
+          await databaseBuilder.commit();
 
-        it('should return the account recovery demand', async () => {
           // when
           const demand = await accountRecoveryDemandRepository.findByTemporaryKey(temporaryKey);
 
           // then
-          expect(demand.length).to.equal(1);
-          expect(demand[0].id).to.equal(demandId);
-          expect(demand[0].temporaryKey).to.equal(temporaryKey);
-          expect(demand[0].used).to.equal(false);
+          expect(demand.id).to.equal(demandId);
+          expect(demand.temporaryKey).to.equal(temporaryKey);
+          expect(demand.used).to.equal(false);
+        });
+
+        it('should throw Account Recovery Demand Expired when temporary key has expired one day ago', async () => {
```

Comment:
Suggestion : 

ça pourrait mériter un context ici 
```suggestion
context('when the temporary key expired a day ago', () => {
        it('should throw Account Recovery Demand Expired', async () => {
```